.github/workflows/turing.yaml

```diff
@@ -36,10 +36,130 @@ env:
   ARTIFACT_RETENTION_DAYS: 7
 
 jobs:
+  build-api:
```

Comment:
Changed the flow of CI to speed it up.
**Before:**
`test -> build -> e2e-test -> publish`
**Now:**
`build -> test & e2e -> publish` 

This reduces the average pipeline execution from 20m to 14m

Reply:
Since `test` has no dependency on `build`, can it be `build & test -> e2e -> publish` ? With this, the expensive e2e can be skipped if the unit tests fail.

Reply:
I thought about this, but ended up with the current setup for two reasons:
* `build & test -> e2e -> publish` pipeline will take  roughly 3.5mins longer to complete, because `build & test` step will require `max(jobs in build&test)` time to complete, which is ~6min, as compared to `max(jobs in build)` time (~2.5min)
* There is no limits on parallel jobs or runner minutes in GitHub Actions, so `expensive` here is not very relevant